pkg/frontend/openshiftpods_get.go

```diff
@@ -0,0 +1,53 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"path/filepath"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+)
+
+func (f *frontend) getOpenShiftPods(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
+	r.URL.Path = filepath.Dir(r.URL.Path)
+
+	b, err := f._getOpenShiftPods(ctx, r, log)
+
+	reply(log, w, nil, b, err)
+}
+
+func (f *frontend) _getOpenShiftPods(ctx context.Context, r *http.Request, log *logrus.Entry) ([]byte, error) {
+	vars := mux.Vars(r)
+
+	namespace := r.URL.Query().Get("namespace")
+
+	err := validateNamespace(r.Method, namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	doc, err := f.dbOpenShiftClusters.Get(ctx, r.URL.Path)
+	switch {
+	case cosmosdb.IsErrorStatusCode(err, http.StatusNotFound):
+		return nil, api.NewCloudError(http.StatusNotFound, api.CloudErrorCodeResourceNotFound, "", "The Resource '%s/%s' under resource group '%s' was not found.", vars["resourceType"], vars["resourceName"], vars["resourceGroupName"])
+	case err != nil:
+		return nil, err
+	}
+
+	k, err := f.kubeActionsFactory(log, f.env, doc.OpenShiftCluster)
+	if err != nil {
+		return nil, err
+	}
+
+	return k.KubeList(ctx, "Pod", namespace)
```

Comment:
This file is called `openshiftpods_get.go` yet you are truly listing pods in a given namespace rather than "getting" a single pod. Could you rename the files to be `openshiftpods_list.go` pattern?

Reply:
It should be protected by rate limiting in the kube actions but i'd be curious to debug that object.

Reply:
New top pods and top nodes functions do implement rate limiting.